File: app/ios/image_toolkit/ui/screen/MergeScreen.swift
import SwiftUI

struct MergeScreen: View {
    @State private var direction: String = "horizontal"
    @State private var inputPath: String = ""
    @State private var outputPath: String = ""
    @State private var spacing: String = "0"
    @State private var gridRows: String = "2"
    @State private var gridCols: String = "2"
    @State private var selectedFormats: Set<String> = []
    @State private var showingAlert = false
    @State private var alertMessage = ""
    
    let directions = ["horizontal", "vertical", "grid"]
    let allFormats = ["jpg", "png", "bmp", "gif", "webp", "tiff"]
    
    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 16) {
                Text("Merge Images")
                    .font(AppTheme.Typography.headlineSmall)
                
                VStack(alignment: .leading, spacing: 8) {
                    Text("Direction")
                        .font(.caption)
                        .foregroundColor(.gray)
                    
                    Picker("Direction", selection: $direction) {
                        ForEach(directions, id: \.self) { dir in
                            Text(dir.capitalized).tag(dir)
                        }
                    }
                    .pickerStyle(MenuPickerStyle())
                    .frame(maxWidth: .infinity)
                    .padding(8)
                    .background(RoundedRectangle(cornerRadius: 8).stroke(Color.gray.opacity(0.2), lineWidth: 1))
                }
                
                FileInput(label: "Input Paths (Files or Dir)", path: $inputPath)
                
                SectionCard(title: "Output Path (Optional)") {
                    FileInput(label: "Output Path", path: $outputPath)
                }
                
                FormatSelector(
                    title: "Input Formats (if input is dir)",
                    formats: allFormats,
                    selectedFormats: $selectedFormats
                )
                
                VStack(alignment: .leading, spacing: 8) {
                    Text("Spacing (px)")
                        .font(.caption)
                        .foregroundColor(.gray)
                    TextField("0", text: $spacing)
                        .keyboardType(.numberPad)
                        .textFieldStyle(RoundedBorderTextFieldStyle())
                        .onChange(of: spacing) { newValue in
                            spacing = newValue.filter { "0123456789".contains($0) }
                        }
                }
                
                if direction == "grid" {
                    HStack(spacing: 16) {
                        VStack(alignment: .leading, spacing: 8) {
                            Text("Rows")
                                .font(.caption)
                                .foregroundColor(.gray)
                            TextField("2", text: $gridRows)
                                .keyboardType(.numberPad)
                                .textFieldStyle(RoundedBorderTextFieldStyle())
                                .onChange(of: gridRows) { newValue in
                                    gridRows = newValue.filter { "0123456789".contains($0) }
                                }
                        }
                        
                        VStack(alignment: .leading, spacing: 8) {
                            Text("Cols")
                                .font(.caption)
                                .foregroundColor(.gray)
                            TextField("2", text: $gridCols)
                                .keyboardType(.numberPad)
                                .textFieldStyle(RoundedBorderTextFieldStyle())
                                .onChange(of: gridCols) { newValue in
                                    gridCols = newValue.filter { "0123456789".contains($0) }
                                }
                        }
                    }
                }
                
                Button(action: {
                    alertMessage = "Running Merge:\nDirection: \(direction)\nInput: \(inputPath)"
                    showingAlert = true
                }) {
                    HStack {
                        Image(systemName: "play.fill")
                        Text("Run Merge")
                    }
                    .frame(maxWidth: .infinity)
                    .padding()
                    .background(AppTheme.primary)
                    .foregroundColor(.white)
                    .cornerRadius(8)
                }
            }
            .padding(16)
        }
        .alert("Merge Action", isPresented: $showingAlert) {
            Button("OK", role: .cancel) { }
        } message: {
            Text(alertMessage)
        }
    }
}

File: app/android/src/main/kotlin/com/personal/image_toolkit/ui/screen/MergeScreen.kt
// ui/screen/MergeScreen.kt
package com.example.imagetoolkit.ui.screen

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.imagetoolkit.ui.components.FileInput
import com.example.imagetoolkit.ui.components.FormatSelector
import android.widget.Toast

@OptIn(ExperimentalMaterial3Api::class, ExperimentalLayoutApi::class)
@Composable
fun MergeScreen() {
    var direction by remember { mutableStateOf("horizontal") }
    val directions = listOf("horizontal", "vertical", "grid")
    var inputPath by remember { mutableStateOf("") }
    var outputPath by remember { mutableStateOf("") }
    var spacing by remember { mutableStateOf("0") }
    var gridRows by remember { mutableStateOf("2") }
    var gridCols by remember { mutableStateOf("2") }
    var selectedFormats by remember { mutableStateOf(setOf<String>()) }
    val allFormats = listOf("jpg", "png", "bmp", "gif", "webp", "tiff")
    val context = LocalContext.current
    var directionExpanded by remember { mutableStateOf(false) }

    LazyColumn(
        modifier = Modifier.fillMaxSize().padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        item { Text("Merge Images", style = MaterialTheme.typography.headlineSmall) }

        item {
            ExposedDropdownMenuBox(
                expanded = directionExpanded,
                onExpandedChange = { directionExpanded = !directionExpanded }
            ) {
                OutlinedTextField(
                    value = direction,
                    onValueChange = {},
                    readOnly = true,
                    label = { Text("Direction") },
                    trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(expanded = directionExpanded) },
                    modifier = Modifier.fillMaxWidth().menuAnchor()
                )
                ExposedDropdownMenu(
                    expanded = directionExpanded,
                    onDismissRequest = { directionExpanded = false }
                ) {
                    directions.forEach { dir ->
                        DropdownMenuItem(
                            text = { Text(dir.capitalize()) },
                            onClick = {
                                direction = dir
                                directionExpanded = false
                            }
                        )
                    }
                }
            }
        }

        item {
            FileInput("Input Paths (Files or Dir)", inputPath) { inputPath = it }
        }

        item {
            com.example.imagetoolkit.ui.components.SectionCard("Output Path (Optional)") {
                FileInput("Output Path", outputPath) { outputPath = it }
            }
        }

        item {
            FormatSelector(
                title = "Input Formats (if input is dir)",
                formats = allFormats,
                selectedFormats = selectedFormats,
                onFormatToggle = { format ->
                    selectedFormats = if (selectedFormats.contains(format))
                        selectedFormats - format else selectedFormats + format
                }
            )
        }

        item {
            OutlinedTextField(
                value = spacing,
                onValueChange = { spacing = it.filter { char -> char.isDigit() } },
                label = { Text("Spacing (px)") },
                keyboardOptions = androidx.compose.ui.text.input.KeyboardOptions(keyboardType = androidx.compose.ui.text.input.KeyboardType.Number),
                modifier = Modifier.fillMaxWidth()
            )
        }

        if (direction == "grid") {
            item {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(16.dp)
                ) {
                    OutlinedTextField(
                        value = gridRows,
                        onValueChange = { gridRows = it.filter { char -> char.isDigit() } },
                        label = { Text("Rows") },
                        keyboardOptions = androidx.compose.ui.text.input.KeyboardOptions(keyboardType = androidx.compose.ui.text.input.KeyboardType.Number),
                        modifier = Modifier.weight(1f)
                    )
                    OutlinedTextField(
                        value = gridCols,
                        onValueChange = { gridCols = it.filter { char -> char.isDigit() } },
                        label = { Text("Cols") },
                        keyboardOptions = androidx.compose.ui.text.input.KeyboardOptions(keyboardType = androidx.compose.ui.text.input.KeyboardType.Number),
                        modifier = Modifier.weight(1f)
                    )
                }
            }
        }

        item {
            Button(
                onClick = {
                    val msg = "Running Merge:\nDirection: $direction\nInput: $inputPath"
                    Toast.makeText(context, msg, Toast.LENGTH_LONG).show()
                },
                modifier = Modifier.fillMaxWidth()
            ) {
                Icon(Icons.Default.PlayArrow, contentDescription = null, modifier = Modifier.padding(end = 8.dp))
                Text("Run Merge", fontSize = 16.sp)
            }
        }
    }
}